signac/core/collection_api.py

```diff
@@ -0,0 +1,491 @@
+                        self._data = self._data[:len(data)]
+                    else:
+                        self.extend(data[len(self):])
+                self.sync()
+            except BaseException:  # rollback
+                self._data = backup
+                raise
+        else:
+            raise ValueError("The data must be a non-string sequence or None.")
+
+    def __delitem__(self, item):
+        self.load()
+        with self._safe_sync():
+            del self._data[item]
+            self.sync()
+
+    def __setitem__(self, key, value):
+        self.load()
+        with self._safe_sync():
+            with self._suspend_sync():
+                self._data[key] = self.from_base(data=value, parent=self)
+            self.sync()
+
+    def __getitem__(self, key):
+        self.load()
+        return self._data[key]
+
+    def __iter__(self):
+        self.load()
+        return iter(self._data)
+
+    def __len__(self):
+        self.load()
+        return len(self._data)
+
+    def __reversed__(self):
+        self.load()
+        return reversed(self._data)
+
+    def __iadd__(self, iterable):
+        self.load()
+        with self._safe_sync():
+            self._data += [self.from_base(data=value, parent=self) for value in iterable]
+            self.sync()
+
+    def __call__(self):
+        self.load()
+        return self.to_base()
+
+    def __eq__(self, other):
+        if isinstance(other, type(self)):
+            return self() == other()
+        else:
+            return self() == other
+
+    def __repr__(self):
+        return repr(self())
+
+    def __str__(self):
+        return str(self())
+
+    def insert(self, index, item):
+        self.load()
+        with self._safe_sync():
+            with self._suspend_sync():
+                self._data.insert(index, self.from_base(data=item, parent=self))
+            self.sync()
+
+    def append(self, item):
+        self.load()
+        with self._safe_sync():
+            with self._suspend_sync():
+                self._data.append(self.from_base(data=item, parent=self))
+            self.sync()
+
+    def extend(self, iterable):
+        self.load()
+        with self._safe_sync():
+            with self._suspend_sync():
+                self._data.extend([self.from_base(data=value, parent=self) for value in iterable])
+            self.sync()
+
+    def remove(self, item):
+        self.load()
+        with self._safe_sync():
+            with self._suspend_sync():
+                self._data.remove(self.from_base(data=item, parent=self))
+            self.sync()
+
+    def clear(self):
+        with self._safe_sync():
+            self._data = []
+            self.sync()
+
+
+class JSONCollection(SyncedCollection):
+
+    def __init__(self, **kwargs):
+        filename = kwargs.pop('filename', None)
+        self._filename = None if filename is None else os.path.realpath(filename)
+        self._write_concern = kwargs.pop('write_concern', True)
```

Comment:
Looks like the default is `False` in the constructor below. Are these different for a reason?

Reply:
Sorry, that's a mistake. Will correct it.